### ISSUE TITLE
guicursor does not seem to work with urxvt

### ISSUE THREAD

**Issue by DarkDefender** [neovim/neovim#6441]
- `nvim --version`: NVIM v0.2.0-1171-g4566f7c
- Vim (version: ) behaves differently? N/A
- Operating system/version: Gentoo linux
- Terminal name/version: urxvt (master) rxvt-unicode (urxvt) v9.22 - released: 2016-01-23
- `$TERM`: rxvt-unicode-256color

### Actual behaviour
The terminal cursor does not change shape anymore when switching between modes. Only the defult terminal cursor shape is shown.

### Expected behaviour
The cursor shape should change like it did with "$NVIM_TUI_ENABLE_CURSOR_SHAPE"

### Steps to reproduce using `nvim -u NORC`
Start nvim with uxrvt, enter insert mode, notice that the cursor shape did not change.

**Comment by justinmk** [comment]
On latest master we take the conservative approach for most terminals: 'guicursor' will be disabled unless we are certain it won't cause problems. If you know your terminal supports cursor shape-changing, just set 'guicursor' explicitly in your init.vim. E.g.:

    set guicursor=n-v-c:block-Cursor/lCursor,i-ci:ver25-Cursor/lCursor,r-cr:hor20-Cursor/lCursor,-blinkwait175-blinkoff150-blinkon175

**Comment by DarkDefender** [comment]
Ah, I see. I thought that it was a problem because it worked out of the box with the VTE based terminals I tested.

However, I think that it should be more clear that it unsets 'guicursor' if it is unsure if the terminal supports it.

I looked at the "Following HEAD" page and ":help guicursor" before opening this ticket and I got the impression that you had to explicitly unset it because it would be always on by default. So I think that perhaps adding a mention that it will be off by default on certain terminals would be good.
